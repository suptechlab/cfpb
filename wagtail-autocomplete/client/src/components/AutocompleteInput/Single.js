import React, { PureComponent } from 'react';

import { nc } from '.';
import { RemoveIcon } from './Icons';
import Suggestions from './Suggestions';


class Single extends PureComponent {
  render() {
    const {
      labelId,
      selected,
      onChange,
      onClick,
      onCreate,
      input,
      canCreate,
    } = this.props;

    if (selected) {
      return (
        <div className={nc('selection', 'selection--single')}>
          <span className={nc('selection__label')}>{selected.title}</span>

          <button
            type="button"
            className={nc('selection__button')}
            onClick={onClick.bind(null, null)}
          >
            <RemoveIcon className={nc('selection__icon')} />

            <span className={nc('sr-only')}>Remove</span>
          </button>
        </div>
      );
    }

    const suggestions = this.props.suggestions.filter(suggestion => {
      if (!selected) {
        return true;
      }
      return suggestion.pk !== selected.pk;
    });

    return (
      <Suggestions
        labelId={labelId}
        suggestions={suggestions}
        onClick={onClick}
        onCreate={onCreate}
        onChange={onChange}
        canCreate={canCreate}
        input={input}
        isSingle={true}
      />
    );
  }
}


export default Single;
