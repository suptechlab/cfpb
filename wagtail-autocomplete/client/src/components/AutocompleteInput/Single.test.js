import React from 'react';
import { shallow } from 'enzyme';

import Single from './Single';


const mockProps = {
  suggestions: [
    { id: 1, title: 'Alice' },
    { id: 2, title: 'Tekisha' },
  ],
  selected: null,
  onChange: jest.fn(),
  onCreate: jest.fn(),
  onClick: jest.fn(),
  input: { value: '' },
  canCreate: true,
};


describe('Single', () => {
  it('exists', () => {
    expect(Single).toBeDefined();
  });

  it('mounts', () => {
    const single = shallow(
      <Single {...mockProps} />
    );
    expect(single).toMatchSnapshot();
  });
});
