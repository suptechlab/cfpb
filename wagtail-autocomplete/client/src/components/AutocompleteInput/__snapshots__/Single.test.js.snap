// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single mounts 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Suggestions
    canCreate={true}
    input={
        Object {
            "value": "",
          }
    }
    isSingle={true}
    labelId={undefined}
    onChange={[Function]}
    onClick={[Function]}
    onCreate={[Function]}
    suggestions={
        Array [
            Object {
              "id": 1,
              "title": "Alice",
            },
            Object {
              "id": 2,
              "title": "Tekisha",
            },
          ]
    }
/>,
  "nodes": Array [
    <Suggestions
      canCreate={true}
      input={
            Object {
                  "value": "",
                }
      }
      isSingle={true}
      labelId={undefined}
      onChange={[Function]}
      onClick={[Function]}
      onCreate={[Function]}
      suggestions={
            Array [
                  Object {
                    "id": 1,
                    "title": "Alice",
                  },
                  Object {
                    "id": 2,
                    "title": "Tekisha",
                  },
                ]
      }
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Single
        canCreate={true}
        input={
                Object {
                        "value": "",
                      }
        }
        onChange={[Function]}
        onClick={[Function]}
        onCreate={[Function]}
        selected={null}
        suggestions={
                Array [
                        Object {
                          "id": 1,
                          "title": "Alice",
                        },
                        Object {
                          "id": 2,
                          "title": "Tekisha",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Single {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "canCreate": true,
          "input": Object {
            "value": "",
          },
          "onChange": [Function],
          "onClick": [Function],
          "onCreate": [Function],
          "selected": null,
          "suggestions": Array [
            Object {
              "id": 1,
              "title": "Alice",
            },
            Object {
              "id": 2,
              "title": "Tekisha",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Suggestions
          canCreate={true}
          input={
                    Object {
                              "value": "",
                            }
          }
          isSingle={true}
          labelId={undefined}
          onChange={[Function]}
          onClick={[Function]}
          onCreate={[Function]}
          suggestions={
                    Array [
                              Object {
                                "id": 1,
                                "title": "Alice",
                              },
                              Object {
                                "id": 2,
                                "title": "Tekisha",
                              },
                            ]
          }
/>,
        "_debugID": 2,
        "_renderedOutput": <Suggestions
          canCreate={true}
          input={
                    Object {
                              "value": "",
                            }
          }
          isSingle={true}
          labelId={undefined}
          onChange={[Function]}
          onClick={[Function]}
          onCreate={[Function]}
          suggestions={
                    Array [
                              Object {
                                "id": 1,
                                "title": "Alice",
                              },
                              Object {
                                "id": 2,
                                "title": "Tekisha",
                              },
                            ]
          }
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Single
    canCreate={true}
    input={
        Object {
            "value": "",
          }
    }
    onChange={[Function]}
    onClick={[Function]}
    onCreate={[Function]}
    selected={null}
    suggestions={
        Array [
            Object {
              "id": 1,
              "title": "Alice",
            },
            Object {
              "id": 2,
              "title": "Tekisha",
            },
          ]
    }
/>,
}
`;
